<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ CDSeq</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CDSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/CDSeq-vignette.html">CDSeq: a complete deconvolution method using sequencing data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kkang7/CDSeq_R_Package/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="CDSeq-R-package.html">CDSeq-R-package</a></code> </p>
        </td>
        <td><p>CDSeq: A package for complete deconvolution using sequencing data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CDSeq.html">CDSeq()</a></code> </p>
        </td>
        <td><p>Complete deconvolution using sequencing data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Cell2RNA.html">Cell2RNA()</a></code> </p>
        </td>
        <td><p>Cell proportion to RNA proportion
<code>Cell2RNA</code> converts Cell proportion to RNA proportion</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RNA2Cell.html">RNA2Cell()</a></code> </p>
        </td>
        <td><p>RNA proportion to cell proportion
<code>RNA2Cell</code> converts RNA proportion to cell proportion</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="SyntheticMixtureData.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Synthetic bulk RNA-seq read counts data of six cell types, PBMC mixtures using scRNASeq and some preliminary results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cdseq.result.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Output of synthetic mixtures of PBMC scRNAseq data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cellTypeAssign.html">cellTypeAssign()</a></code> </p>
        </td>
        <td><p>Assign cell types using correlation matrix computed using cell-type-specific GEPs and reference GEPs.
<code>cellTypeAssign</code> assigns CDSeq-identified cell types to reference profile.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cellTypeAssignMarkerGenes.html">cellTypeAssignMarkerGenes()</a></code> </p>
        </td>
        <td><p><code>cellTypeAssignMarkerGenes</code> assigns CDSeq-identified cell types using user-provided marker gene list and plots heatmap.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cellTypeAssignSCRNA.html">cellTypeAssignSCRNA()</a></code> </p>
        </td>
        <td><p><code>cellTypeAssignSCRNA</code> assigns CDSeq-identified cell types using single cell RNAseq data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gene2rpkm.html">gene2rpkm()</a></code> </p>
        </td>
        <td><p>gene2rpkm outputs the rpkm normalizations of the CDSeq-estimated GEPs.
<code>gene2rpkm</code> outputs the rpkm normalizations of the CDSeq-estimated GEPs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gene_length.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gibbsSampler.html">gibbsSampler()</a></code> </p>
        </td>
        <td><p>This is the Gibbs sampler for CDSeq.
<code>GibbsSampler</code> returns estimated GEPs and cell type proportions.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hungarian_Rcpp.html">hungarian_Rcpp()</a></code> </p>
        </td>
        <td><p>This is the Hungarian algorithm wrapper for cell type assignment
<code>hungarian_Rcpp</code> returns cell type assignment given reference GEPs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="intersection.html">intersection()</a></code> </p>
        </td>
        <td><p><code>intersection</code> take intersection of multiple lists and return the common set and index</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="logpost.html">logpost()</a></code> </p>
        </td>
        <td><p>logpost computes the log posterior of the CDSeq model.
<code>logpost</code> outputs the value of log posterior.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="max_rep.html">max_rep()</a></code> </p>
        </td>
        <td><p><code>max_rep</code> Find the element that repeats the most in a given vector and calculate its proportion.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge_df.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Data frame for keeping the CDSeq-estimated cell type proportions for PBMC mixtures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mixtureGEP.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Synthetic bulk RNA-seq read counts data of six cell types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbmc_ggplot.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>ggplot figures of comparison between CDSeq-estimated cell type proportion and ground truth</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbmc_mix.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Synthetic bulk RNA-seq read counts data of PBMC single cell data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read2gene.html">read2gene()</a></code> </p>
        </td>
        <td><p>read2gene outputs the GEP normalized by gene length of the CDSeq-estimated GEPs.
<code>read2gene</code> outputs the gene length normalized CDSeq-estimated GEP.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="refGEP.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>GEPs of six component pure cell lines</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="result1.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>CDSeq result of synthetic bulk RNA-seq read counts data of six cell types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="result2.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>CDSeq result of synthetic bulk RNA-seq read counts data of six cell types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="result3.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>CDSeq result of synthetic bulk RNA-seq read counts data of six cell types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sc_annotation.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>Cell type annotation of the PBMC single cell data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sc_gep.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>PBMC single cell RNAseq read counts that used for creating synthetic PBMC mixtures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seedMT.html">seedMT()</a></code> </p>
        </td>
        <td><p>This is the Mersenne Twister random number generator.
<code>cokus</code> generates pseudorandom integers uniformly distributed in 0..(2^32 - 1).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_GEP_gene.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True GEPs of the six component cell types normalized by gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_GEP_read.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True GEPs of the six component cell types unnormalized by gene length</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_GEP_rpkm.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True GEPs of the six component cell types RPKM normalization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_prop.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True cell type proportion in the PBMC synthetic mixtures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_prop_RNA.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True cell type RNA proportions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="true_prop_cell.html">SyntheticMixtureData</a></code> </p>
        </td>
        <td><p>True cell proportions of the mixtures</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Kai Kang, Caizhi Huang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


